modules/installation-special-config-kmod.adoc

```diff
@@ -0,0 +1,304 @@
+
+=== Provision kernel modules at cluster install time
+
+First create a base Ignition config that you'd like to use. It will
+contain the ssh public key to add to the authorized_keys file for
+the core user and also a systemd unit file. The systemd unit file
+require-simple-kmod.service that requires kmods-via-containers@simple-kmod.service.
+
+The systemd unit is a workaround for an
+link:https://github.com/coreos/ignition/issues/586[upstream bug]
+and makes sure that the kmods-via-containers@simple-kmod.service gets started
+on boot:
+
+. Create an Ignition config file that creates a systemd unit file:
++
+----
+$ cat <<EOF > ./baseconfig.ign
+{
+  "ignition": { "version": "3.0.0" },
+  "passwd": {
+    "users": [
+      {
+        "name": "core",
+        "groups": ["sudo"],
+        "sshAuthorizedKeys": [
+          "ssh-rsa AAAA"
+        ]
+      }
+    ]
+  },
+  "systemd": {
+    "units": [{
+      "name": "require-kvc-simple-kmod.service",
+      "enabled": true,
+      "contents": "[Unit]\nRequires=kmods-via-containers@simple-kmod.service\n[Service]\nType=oneshot\nExecStart=/usr/bin/true\n\n[Install]\nWantedBy=multi-user.target"
+    }]
+  }
+}
+EOF
+----
++
+[NOTE]
+====
+You'll need to add your public SSH key to that baseconfig.ign.
+====
+
+. Create a fakeroot directory and populate it with files that we want to
+deliver via Ignition, using the repositories cloned earlier:
++
+----
+$ FAKEROOT=$(mktemp -d)
+$ cd ../kmods-via-containers
+$ make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+$ cd ../kvc-simple-kmod
+$ make install DESTDIR=${FAKEROOT}/usr/local CONFDIR=${FAKEROOT}/etc/
+----
+
+. Use a tool call the `filetranspiler` to generate a final Ignition config (`config.ign`),
+and have it include the base Ignition config and the fakeroot directory with files you
+would like to deliver:
++
+----
+$ cd ..
+$ git clone https://github.com/ashcrow/filetranspiler
+$ ./filetranspiler/filetranspile -i ./baseconfig.ign \
+     -f ${FAKEROOT} -p -o config.ign
+----
+
+. Use this ignition config (`config.ign`) to start a RHEL CoreOS node and
+see the kmods-via-containers@simple-kmod.service and the kernel modules
+associated with simple-kmods get loaded. You can check the modules are
+loaded by going on to node (using `oc debug`) and running this command:
++
+----
+$ lsmod | grep simple_
+simple_procfs_kmod     16384  0
+simple_kmod            16384  0
+----
+
+=== Provision kernel modules via the MCO
+
+. Start with a base MCO yaml snippet that looks like:
++
+----
+$ cat <<EOF > mc-base.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 10-kvc-simple-kmod
+spec:
+  config:
+EOF
+----
+
+. Start with a base ignition config snippet that looks like the following. Note that you do not need an SSH key here because the OpenShift Container Platform cluster install already has configs for the SSH key defined.
```

Comment:
Ignition

Reply:
s/Note that you/You do not 
s/ OpenShift Container Platform/{product-title}
s/install/installation

Reply:
Done